### PR TITLE
lwp inline refactor

### PR DESCRIPTION
**Pull request by muff1n1634**
Have source files include inline functions directly instead of including them through library headers
see #137, [this comment](https://github.com/devkitPro/libogc/pull/137#issuecomment-1405324147)

things to do

- [x] move `__lwp_mutex_seize_irq_trylock` and `__lwp_mutex_seize` out to `lwp_mutex.inl`
- [x] make `_seize` `static __inline__` instead of a macro (? not sure if there's a specific reason it's a macro)
- [x] include inline source files in the other source files that need to use them
- [x] remove `LIBOGC_INTERNAL` ifdef from headers (and remove -D flag from `CFLAGS` in makefile)
- [x] for those files only including these headers for the inline functions, remove those includes*
- [x] remove personal config files (sublime-project, entry in .gitignore) (sorry i track these by habit)
- [x] clean up

road bumps

- [x] code outside of libogc: compiler cannot find inline files. figure that out ~tomorrow~ ~later~

### CODE REVIEW

File: libogc/exi.c [hunk 1]
```diff
@@ -38,6 +38,8 @@ distribution.
 #include "exi.h"
 #include "gcutil.h"
 
+#include "lwp_queue.h"
```
Comment (DacoTaco) [inline libogc/exi.c:41]: how did this header get included before? :/

**Review by DacoTaco — CHANGES_REQUESTED** [review]
its still in your todo, but the sublime files are indeed not wanted hehe :D
other than that, looks good so far. i will have to test it all when youre done

File: libogc/exi.c (hunk 1 above)
Comment (muff1n1634) [reply libogc/exi.c]: typo haha

**Comment by DacoTaco** [comment]
` code outside of libogc: compiler cannot find inline files. figure that out tomorrow`
got an example? code outside of libogc should not be using those inline functions/inl files?
if they do, we need to have a plan for them

**Comment by muff1n1634** [comment]
as an example, [`libdb/debug.c`](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/libdb/debug.c#L619) has references to [`__lwp_thread_dispatchdisable`](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/libdb/debug.c#L626) and [`__lwp_thread_dispatchenable`](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/libdb/debug.c#L651), inlines from `lwp_threads.inl` (lines [43](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/libogc/lwp_threads.inl#L43) and [37](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/libogc/lwp_threads.inl#L37))

the other offenders i found are in libtinysmb ([`smb.c`](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/libtinysmb/smb.c#L394)), libwiikeyboard ([`keyboard.c`](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/libwiikeyboard/keyboard.c#L556)), lwbt ([`bte.h`](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/lwbt/bte.c#L135), lwip ([`network.c`](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/lwip/network.c#L173), [`netif/loopif.c`](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/lwip/netif/loopif.c#L105)), and wiiuse ([`io.c`](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/wiiuse/io.c#L32), [`io_wii.c`](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/wiiuse/io_wii.c#L160), [`motion_plus.c`](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/wiiuse/motion_plus.c#L25), [`wiiuse.c`](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/wiiuse/wiiuse.c#L67), [`wpad.c`](https://github.com/devkitPro/libogc/blob/4ec9a84931534b9ec2aaa8f8553be44b27950d42/wiiuse/wpad.c#L140))

these should be all the files that have at least one inline function, but some may have more than one

**Comment by DacoTaco** [comment]
could you access them through `extern __lwp_thread_dispatchenable ...` ?

**Comment by muff1n1634** [comment]
it compiles with externs, but my initial thoughts on linking time are that it's not going to work because they're declared static

i'm going to test it right now though, i'll update as i go

**Comment by muff1n1634** [comment]
yeah, linker throws an error about `undefined reference to __lwp_thread_dispatchenable` when i added extern def to it (tested with libdb)

**Comment by muff1n1634** [comment]
i mean, worst case scenario we just inline them manually? that seems pretty ugly to me though, i would rather not have to do that

**Comment by DacoTaco** [comment]
hm, inlining them manually is a no go. that just duplication shit. a relativepath include would work, but is not pretty either. 

if external libs outside of libogc use it, maybe it should just be a regular function?

**Comment by muff1n1634** [comment]
like adding all of the inlines into the headers instead of having a separate inline file for them? or just the ones that the other libraries are using? or something else

**Comment by DacoTaco** [comment]
i wouldn't make *all* the inlines public. just the ones libraries in this repo besides libogc use.

**Comment by WinterMute** [comment]
The other libraries built alongside libogc and distributed as part of the libogc package are somewhat privileged in that they can access libogc internal headers. They probably shouldn't but for now this particular issue is probably best solved by adding the path to those .inl headers to the includes for this project. Nothing else can ever have used these functions since the headers aren't distributed.

Will need looking at again later I think but this is probably the simplest way to go for now.

```diff
diff --git a/Makefile b/Makefile
index 3ba48f2..589c7f6 100644
--- a/Makefile
+++ b/Makefile
@@ -76,7 +76,7 @@ WIIKEYBLIB    := $(LIBDIR)/libwiikeyboard
 STUBSLIB       := $(LIBDIR)/libgclibstubs
 
 #---------------------------------------------------------------------------------
-DEFINCS                := -I$(BASEDIR) -I$(BASEDIR)/gc
+DEFINCS                := -I$(BASEDIR) -I$(BASEDIR)/gc -I$(OGCDIR)
 INCLUDES       :=      $(DEFINCS) -I$(BASEDIR)/gc/netif -I$(BASEDIR)/gc/ipv4 \
                                -I$(BASEDIR)/gc/ogc -I$(BASEDIR)/gc/ogc/machine \
                                -I$(BASEDIR)/gc/modplay \
diff --git a/lwbt/bte.c b/lwbt/bte.c
index d942ee7..cfc33c8 100644
--- a/lwbt/bte.c
+++ b/lwbt/bte.c
@@ -13,7 +13,7 @@
 #include "btmemb.h"
 #include "physbusif.h"
 
-#include "lwp_thread.inl"
+#include "lwp_threads.inl"
 
 #define STACKSIZE                                              32768
 #define MQ_BOX_SIZE                                            256
 ```

File: lwbt/bte.c [hunk 2]
```diff
@@ -13,7 +12,7 @@
 #include "btmemb.h"
 #include "physbusif.h"
 
-#include "lwp_thread.inl"
```
Comment (muff1n1634) [inline lwbt/bte.c]: grrr another typo

**Comment by muff1n1634** [comment]
build is ok

i also did a final pass through everything i edited on this branch, looked ok to me

File: libogc/lwp_heap.inl [hunk 3]
```diff
@@ -1,6 +1,8 @@
 #ifndef __LWP_HEAP_INL__
 #define __LWP_HEAP_INL__
 
+#include "lwp_heap.h"
```
Comment (DacoTaco) [inline libogc/lwp_heap.inl:4]: why is the include of the header here, and not where the inl is included? (like in [lwp_heap.c](https://github.com/devkitPro/libogc/pull/138/files#diff-cd8dda5bd864cfde73409b8ea8b2babccbe5ab5bba93cddd7f338f8ec397639a), it should keep the include of lwp_heap.h, but also include this inline.

this applies for all includes of these inline functions

**Review by DacoTaco — CHANGES_REQUESTED** [review]
this is the only remark i have. other than that it looks good to me

EDIT: nvm, its better this way.

**Review by DacoTaco — APPROVED** [review]

**Comment by muff1n1634** [comment]
crap, i forgot to add the numbers to the commit email, now it looks kinda weird :/

**Comment by DacoTaco** [comment]
wdym?

**Pull request merged**